### PR TITLE
Linked list attempt

### PR DESCRIPTION
**Pull request by wiggity**

### CODE REVIEW

File: course3/homework-linkedlist/my_linked_list.h [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+			for (iterator it = begin(); it != end(); ++it) {
+				if (it.mPtr->mNext.get() == toErase.mPtr) {
+					it.mPtr->mNext = std::move(toErase.mPtr->mNext);
+					numNodes--;
+					break;
+				}
+			}
+		}
+	private:
+		std::size_t numNodes;
```
Comment (maciekgajewski) [inline course3/homework-linkedlist/my_linked_list.h]: inconsistent naming

File: course3/homework-linkedlist/test.cc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+	assert(list1.front() == 2);
+
+	list1.pop_front();
+	assert(list1.empty());
+	assert(list1.size() == 0);
+
+	list1.push_front(3);
+	list1.push_front(4);
+	list1.push_front(5);
```
Comment (maciekgajewski) [inline course3/homework-linkedlist/test.cc:28]: Would be nice being able to say:

```
list1 = {5, 4, 3};
```

File: course3/homework-linkedlist/test.cc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+int main()
+{
+	// test fior std::string - works
+	std::cout << "Testing std::list<int>..." << std::endl;
+	list_test<std::list<int>>();
+	std::cout << "std::list test passes" << std::endl;
+	
+	std::cout << "Testing MyLinkedList..." << std::endl;
+	list_test<MyLinkedList<int>>();
```
Comment (maciekgajewski) [inline course3/homework-linkedlist/test.cc:58]: Your list has pretty strict type requirement: the stored type must be copyable. I propose the following excercise:
1. Allow for storing move-only types (MyLinkedListstd::unique_ptr<int>)
2. Allow for storing types that don't have neither copy nor move constructor, only initializing constructor.

File: course3/homework-linkedlist/my_linked_list.h [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+template<typename T>
+class MyLinkedList {
+	struct Node {
+		T mData;
+		std::unique_ptr<Node> mNext;
+		Node(const T& value) : mData(value) {}
+	};
+
+	public:
```
Comment (maciekgajewski) [inline course3/homework-linkedlist/my_linked_list.h:14]: There is too much indentation, and it's inconsistent (Node is indented once, iterator twice)

File: course3/homework-linkedlist/test.cc (hunk 2 above)
Comment (wiggity) [inline course3/homework-linkedlist/test.cc:28]: done.

File: course3/homework-linkedlist/my_linked_list.h (hunk 4 above)
Comment (wiggity) [inline course3/homework-linkedlist/my_linked_list.h:14]: Fixed indentation.

File: course3/homework-linkedlist/my_linked_list.h (hunk 1 above)
Comment (wiggity) [inline course3/homework-linkedlist/my_linked_list.h]: Fixed.

File: course3/homework-linkedlist/test.cc (hunk 2 above)
Comment (maciekgajewski) [inline course3/homework-linkedlist/test.cc:28]: push it then, let me see it!

Comment (wiggity) [inline course3/homework-linkedlist/test.cc:28]: pushed already! see here for constructor: https://github.com/maciekgajewski/c-excercises/pull/40/files#diff-54782b73840c4c7e336f56991bf8f2dcR66

and here for the driver: https://github.com/maciekgajewski/c-excercises/pull/40/files#diff-05233081ce776646e2e00d63628259c8R45

**Pull request closed without merging**